lib/basevm.py

```diff
@@ -13,14 +13,25 @@ class BaseVM:
     """ Base class for VM and Snapshot. """
 
     xe_prefix = "vm"
+    uuid: str
 
-    def __init__(self, uuid, host: 'lib.host.Host'):
+    def __init__(self, uuid: str, host: 'lib.host.Host'):
         logging.info("New %s: %s", type(self).__name__, uuid)
         self.uuid = uuid
         self.host = host
 
+    @overload
+    def param_get(self, param_name: str, key: Optional[str] = ...,
+                  accept_unknown_key: Literal[False] = ...) -> str:
+        ...
+
+    @overload
+    def param_get(self, param_name: str, key: Optional[str] = ...,
+                  accept_unknown_key: bool = ...) -> Optional[str]:
+        ...
+
     def param_get(self, param_name: str, key: Optional[str] = None,
-                  accept_unknown_key: bool = False) -> Union[str, bool, None]:
+                  accept_unknown_key: bool = False) -> Optional[str]:
```

Comment:
There shouldn't be a need to keep any type hints here, since everything is in the overloads now, ... or is there?

Reply:
I tried to remove them but then pyright is not detecting the type properly in the IDE (while not complaining when running from the command line)

Reply:
oh. That could be an issue to raise to the pyright project?

Reply:
Ok, it should be better, removed the type hint on the real function, the `lib/common.py::_param_get` has a fall back for bool that is working on my own editor pyright.

Reply:
your pyright is able to find the type of `param_name`, where it fails to do that in the screenshot above?

Reply:
Nope, not this, I misunderstood your problem.

Reply:
ok, too bad, it would have been better with just the annotations on the overloads :)
We can maybe work on Pyright to make it work, but in the meantime, please keep the type annotations on the functions args: that's definitely some useful info!